src/org/openlcb/OlcbInterface.java

```diff
@@ -271,8 +282,23 @@ public void registerStartNotification(ConnectionListener c) {
      */
     public void dispose(){
         // shut down the thread pool
-        if(threadPool != null) {
-           threadPool.shutdownNow();
+        if(threadPool != null && !(threadPool.isShutdown())) {
+           // modified from the javadoc for ExecutorService 
+           threadPool.shutdown(); // Disable new tasks from being submitted
+           try {
+              // Wait a while for existing tasks to terminate
+              if (!threadPool.awaitTermination(10, TimeUnit.SECONDS)) {
```

Comment:
I think this delay is causing problems. Here are timings from 'ant run' that runs the tests:
Before your PR:
BUILD SUCCESSFUL
Total time: 8 seconds

After your PR:
BUILD SUCCESSFUL
Total time: 3 minutes 8 seconds

The problem is that as far as I can see shutdown() does not actively interrupt the tasks, and there are tasks that are waiting for interruptedexception in order to terminate. Therefore we will deterministically take 10 seconds to exit.

Maybe we should try shutdownNow() straight away. It is meaningless to wait until we have actually sent out some notification that threads are requested to terminate.